buckets/redis/bucket_factory.go

```diff
@@ -164,6 +168,64 @@ func (bf *bucketFactory) reconnectToRedis(oldClient *redis.Client) {
 	}
 }
 
+func (bf *bucketFactory) handleConnectionFailure(oldClient *redis.Client) {
+	bf.Lock()
+	defer bf.Unlock()
+
+	if oldClient == bf.client && !bf.connectionNeedsResolution {
+		logging.Print("Attempting to establish new connection to redis")
+		bf.connectionNeedsResolution = true
+		go bf.establishNewConnectionToRedis(oldClient)
+	}
+}
+
+func (bf *bucketFactory) establishNewConnectionToRedis(oldClient *redis.Client) {
+	client := oldClient
+	numsTried := 0
+	exponentialDelay := 1 * time.Second
+	exponentialDelayCeiling := 5 * time.Minute
```

Comment:
good for now, but we probably want to make it configurable in a follow up PR

Reply:
maybe. It would definitely be useful, but i'm not sure how much it will get used. 

I do think 5 min ceiling is too high. 1 min sounds more reasonable.